src/utils/hooks/useSubmitApplication.tsx

```diff
@@ -1,65 +1,36 @@
-import { useEffect, useState } from 'react'
-import {
-  ApplicationResponse,
-  useGetElementsAndResponsesQuery,
-  useUpdateApplicationMutation,
-} from '../generated/graphql'
-import { UseGetApplicationProps } from '../types'
+import { useState } from 'react'
+import { useUpdateApplicationMutation } from '../generated/graphql'
+import { ResponseFull, UseGetApplicationProps } from '../types'
 
 const useSubmitApplication = ({ serialNumber }: UseGetApplicationProps) => {
   const [submitted, setSubmitted] = useState(false)
   const [processing, setProcessing] = useState(false)
   const [error, setError] = useState('')
 
-  // Hook to get existing responses in cache - triggered when user submits application
-  const { data, error: responsesError } = useGetElementsAndResponsesQuery({
-    variables: {
-      serial: serialNumber,
-    },
-    skip: !submitted,
-    fetchPolicy: 'cache-only',
-  })
-
   const [applicationSubmitMutation] = useUpdateApplicationMutation({
     onCompleted: () => {
       setProcessing(false)
+      console.log('Finished submission')
     },
     onError: (submissionError) => {
       setProcessing(false)
       setError(submissionError.message)
     },
   })
 
-  useEffect(() => {
-    if (responsesError) {
-      setProcessing(false)
-      setError(responsesError.message)
-    }
-    if (
-      !data?.applicationBySerial?.applicationResponses ||
-      data?.applicationBySerial?.applicationResponses.nodes.length === 0
-    )
-      return
-
-    // Transform in array to be included in update patch
-    const responses = data?.applicationBySerial?.applicationResponses.nodes as ApplicationResponse[]
+  const submit = (responses: ResponseFull[]) => {
+    setSubmitted(true)
+    setProcessing(true)
     const responsesPatch = responses.map((response) => {
-      return { id: response.id, patch: { value: response?.value } }
+      return { id: response.id, patch: { value: response.text } } // TODO: Check if passing text for value is ok
```

Comment:
Yes, good point. It was a bit confusing because we don't have those possible other properties listing in `ResponseFull`, I have changed this now on 864b3e5. See what you think.